indexing-service/src/test/java/org/apache/druid/indexing/overlord/sampler/InputSourceSamplerTest.java

```diff
@@ -1060,6 +1062,107 @@ public void testWithFilter() throws IOException
     );
   }
 
+  @Test
+  public void testIndexParseException() throws IOException
```

Comment:
Thanks for adding this test! Similar to https://github.com/apache/druid/pull/10383/files#diff-ef25ac1cc1f275b47b939b65e1d0c8b8b8512aeada52d06b8541b8f381df03eeR2731, could you please add a unit test for sampling a block of multiple JSON strings? The unit test can be run only when `parserType` is `STR_JSON`. [We usually just return in the unit test when the parameter is not what we want to test with](https://github.com/apache/druid/blob/master/processing/src/test/java/org/apache/druid/query/groupby/GroupByQueryRunnerTest.java#L6812-L6814). It would be nice if the test verifies the followings:
- The sampler response when there is no parseException with a list of multiple JSON strings.
- The sampler response when there is a parseException thrown while parsing a list of multiple JSON strings. Maybe you can improve this unit test to do it as well.

Reply:
A block of multiple json strings is only supported in stream ingestion as we discussed above, that's also why the test cases are added in kafka-indexing-service module.[It's implemented by setting the new property on `JsonInputFormat` in `RecordSupplierInputSource`](https://github.com/apache/druid/pull/10383/files#diff-72b4a411d4ce193935a708df516b00f94827e9563ba7adf00c9e0be479435b5bR106).

 For test cases for `InputSourceSampler` here, the input source is `InlineInputSource` which always uses `InputEntityIteratingReader` even when input format is `JsonInputFormat`, there's no chance to test that. So I think we don't need add test cases here.

Reply:
I think we need such tests because `KafkaIndexTaskTest` only covers the actual indexing side, but not the sampling side. Without those tests, we could break the sampler without being noticed.

> For test cases for `InputSourceSampler` here, the input source is `InlineInputSource` which always uses `InputEntityIteratingReader` even when input format is `JsonInputFormat`, there's no chance to test that. So I think we don't need add test cases here.

Maybe this is not a good place for the new tests, but you don't have to use the `InlineInputSource` for them. Rather, you should use `RecordSupplierInputSource` as you mentioned. Since `RecordSupplierInputSource` accepts the `RecordSupplier` interface as a parameter which is the data supplier, I think you could implement a mock for testing (I don't think we should test `Firehose` as it's deprecated). Or do you see some reason you cannot? If it's hard, I'm also OK with adding them in a follow-up.

Reply:
> I think we need such tests because `KafkaIndexTaskTest` only covers the actual indexing side, but not the sampling side. Without those tests, we could break the sampler without being noticed.

I forgot that test cases in `KafkaIndexTaskTest` only cover the indexing process. It's OK for me to add such tests. 

BTW, integration tests failed, I don't think it's related to changes in this PR, and the failure can also be seen in my other PR, is there any other problem ?

Reply:
Thanks for understanding. Do you want to add those tests in this PR? For the integration test failure, I don’t think it’s relevant to the changes in this PR. I just retriggered it.

Reply:
I've added a test [`testMultipleJsonStringInOneBlock`](https://github.com/apache/druid/pull/10383/files#diff-b4114cca5866803474d2b6ec21a1d566b90eab5d4a31d4f051b93cd8883914ebR1187),  please check it.